### ISSUE TITLE
theano.sandbox.cuda deprecated and will be removed

### ISSUE THREAD

**Issue by HenryJia** [fchollet/keras#5209]
Keras' theano backend needs updating

Currently theano with Keras triggers the following warning with the latest master branch of Theano

WARNING (theano.sandbox.cuda): The cuda backend is deprecated and will be removed in the next release.  Please switch to the gpuarray backend. You can get more information about how to switch at this URL:
 https://github.com/Theano/Theano/wiki/Converting-to-the-new-gpu-back-end%28gpuarray%29

**Comment by gvtulder** [comment]
This is related to the [old batch normalization code](https://github.com/fchollet/keras/blob/1585b8dd4e877e022e3c5f5dbecec51666b2218c/keras/backend/theano_backend.py#L484-L568), which is the only bit in Keras that still makes direct calls to the gpu functions. Keras already has [a new implementation](https://github.com/fchollet/keras/blob/1585b8dd4e877e022e3c5f5dbecec51666b2218c/keras/backend/theano_backend.py#L451-L481) that solves this by using a future interface in Theano. This interface is not included in Theano yet, but is waiting in this pull request: Theano/Theano#5190. I think the plan was to include this before the final 0.9.0 release of Theano.

So while it's possible to change the Keras backend to use the `theano.gpuarray` functions instead of `theano.sandbox.cuda`, this would be a temporary fix (hopefully).

**Comment by gvtulder** [comment]
The new batch normalization functions are now available in the Theano master branch, so you should no longer see the warning (if you're running Theano with the `gpuarray` backend).
